### PR TITLE
allow returning metrics for every stage (not only compile+run)

### PR DESCRIPTION
**Pull request by PhilippvK**

### CODE REVIEW

File: mlonmcu/flow/backend.py [hunk 1]
```diff
@@ -72,8 +74,26 @@ def load_model(self, model):
         pass
 
     @abstractmethod
+    def _generate_code(self):
```
Comment (rafzi) [inline mlonmcu/flow/backend.py:77]: maybe we should document the return types. they are not quite clear to me. in the below code, sometimes the underscore version and sometimes the non-underscore version returns a tuple.

**Review by rafzi — APPROVED** [review]
looks good!

File: mlonmcu/flow/backend.py (hunk 1 above)
Comment (PhilippvK) [reply mlonmcu/flow/backend.py:77]: good point, i will look into that. I was going to refactor the naming of these `generate_` and `export_` methods anyway to be more consistent for all types of components.

Comment (PhilippvK) [reply mlonmcu/flow/backend.py]: Refactored the naming and added type hints

**Pull request merged**